### PR TITLE
[boto] Default to None if no region 

### LINKED ISSUE #525: boto3 patch method crashes while fetching AWS region
The `boto3` patch method of the `ddtrace` library expects the boto3 request to have the region parameter set.

Some of requests do not have it set (shared an example with Datadog engineers privately) so the patched function crashes on the following line:

https://github.com/DataDog/dd-trace-py/blob/7a3d0f69610a160bfd71cec8caa8366bc6977a8e/ddtrace/contrib/boto/patch.py#L68

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
Fixes #525, also applies formatting.

### CODE REVIEW

File: ddtrace/contrib/boto/patch.py [hunk 1]
```diff
@@ -47,31 +63,36 @@ def patched_query_request(original_func, instance, args, kwargs):
...
+        region = getattr(instance, "region", None)
         region_name = get_region_name(region)
 
         meta = {
-            'aws.agent': 'boto',
-            'aws.operation': operation_name,
-            'aws.region': region_name,
+            "aws.agent": "boto",
+            "aws.operation": operation_name,
+            "aws.region": region_name,
```
Comment (palazzem) [inline ddtrace/contrib/boto/patch.py]: with the `None` default, `get_region_name(region)` returns `None`. Maybe we should add `aws.region` tag only if it's not none?

Comment (Kyle-Verhoog) [reply ddtrace/contrib/boto/patch.py]: I'd prefer this behaviour, but I opted to stay consistent with the functionality of line 143:
https://github.com/DataDog/dd-trace-py/blob/56cb63f2bc16a70bfe08174f751d4453afcc3566/ddtrace/contrib/boto/patch.py#L143

Comment (Kyle-Verhoog) [reply ddtrace/contrib/boto/patch.py]: This will result in the region tag showing up as `'None'`, which isn't ideal. Let's not add it if it's none.

Comment (palazzem) [reply ddtrace/contrib/boto/patch.py]: exactly, let's not add it if it's `None`

**Comment by Kyle-Verhoog** [comment]
Giving it a quick look, the `moto` library we use to mock `boto` doesn't seem to allow us to mock the region not being set. Creating a regression test would require a non-trivial amount of work.

File: ddtrace/contrib/boto/patch.py [hunk 2]
```diff
@@ -86,14 +86,15 @@ def patched_query_request(original_func, instance, args, kwargs):
...
-        region_name = get_region_name(region)
+        region_name = _get_instance_region_name(instance)
 
         meta = {
             "aws.agent": "boto",
             "aws.operation": operation_name,
-            "aws.region": region_name,
         }
+        if region_name:
+            meta["aws.region"] = region_name
```
Comment (palazzem) [inline ddtrace/contrib/boto/patch.py]: can you move these as a constants? better in a `constants.py` module. So they're all in one place + you use the same variable in 2 parts of the code

**Pull request merged**